server/features/destination_groups.feature

```diff
@@ -0,0 +1,213 @@
+    Then we get deleted response
+
+
+  @auth
+  Scenario: Failure to delete Destination Group referenced by another Destination Group
+    Given empty "destination_groups"
+    When we get "/destination_groups"
+    Then we get list with 0 items
+    When we post to "/destination_groups" with "destgroup1" and success
+    """
+    [
+      {
+        "name":"Group 1", "description": "new stuff",
+        "destination_groups": [], "output_channels": []
+      }
+    ]
+    """
+    Then we get existing resource
+    """
+      {
+        "_id":"#destgroup1#",
+        "name":"Group 1", "description": "new stuff",
+        "destination_groups": [], "output_channels": []
+      }
+    """
+    When we post to "/destination_groups" with "destgroup2" and success
+    """
+    [{
+      "name":"Group 2", "description": "new stuff",
+      "destination_groups": [{"group":"#destgroup1#"}],
+      "output_channels": []
+    }]
+    """
+    Then we get existing resource
+    """
+    {
+      "name":"Group 2", "description": "new stuff",
+      "destination_groups": [{"group":"#destgroup1#"}],
+      "output_channels": []
+    }
+    """
+    When we delete "/destination_groups/#destgroup1#"
+    Then we get error 412
+
+  @auth @test
+  Scenario: Failure to have self referenced Destination Group
+    Given empty "destination_groups"
+    When we get "/destination_groups"
+    Then we get list with 0 items
+    When we post to "/destination_groups" with "destgroup1" and success
+    """
+    [
+      {
+        "name":"Group 1", "description": "new stuff",
+        "destination_groups": [], "output_channels": []
+      }
+    ]
+    """
+    Then we get existing resource
+    """
+      {
+        "_id":"#destgroup1#",
+        "name":"Group 1", "description": "new stuff",
+        "destination_groups": [], "output_channels": []
+      }
+    """
+    When we post to "/destination_groups" with "destgroup2" and success
+    """
+    [{
+      "name":"Group 2", "description": "new stuff",
+      "destination_groups": [{"group":"#destgroup1#"}],
+      "output_channels": []
+    }]
+    """
+    Then we get existing resource
+    """
+    {
+      "name":"Group 2", "description": "new stuff",
+      "destination_groups": [{"group":"#destgroup1#"}],
+      "output_channels": []
+    }
+    """
+    When we post to "/destination_groups" with "destgroup3" and success
+    """
+    [{
+      "name":"Group 3", "description": "new stuff",
+      "destination_groups": [{"group":"#destgroup2#"}],
+      "output_channels": []
+    }]
+    """
+    Then we get existing resource
+    """
+    {
+      "name":"Group 3", "description": "new stuff",
+      "destination_groups": [{"group":"#destgroup2#"}],
+      "output_channels": []
+    }
+    """
+    When we patch "/destination_groups/#destgroup1#"
+    """
+    {
+      "name":"Group 1 testing",
+      "destination_groups": [{"group":"#destgroup2#"}]
+    }
+    """
+    Then we get error 400
```

Comment:
I think we should check the returned error message so it's not some generic message.